zzzz_modular_occulus/code/game/machinery/dorms_elevator.dm

```diff
@@ -0,0 +1,131 @@
+
+	var/cooldown = 24 SECONDS
+	var/cooldown_timer
+	var/list/preserve_items = list(
+		/obj/item/weapon/hand_tele,
+		/obj/item/weapon/card/id/captains_spare,
+		/obj/item/device/aicard,
+		/obj/item/device/mmi,
+		/obj/item/device/paicard,
+		/obj/item/weapon/gun,
+		/obj/item/weapon/pinpointer,
+		/obj/item/clothing/suit,
+		/obj/item/clothing/shoes/magboots,
+		/obj/item/blueprints,
+		/obj/item/clothing/head/space,
+		/obj/item/weapon/storage/internal
+	)
+	var/obj/item/device/radio/intercom/announce
+	var/on_store_message = "has clocked out of their shift and returned to the dormitories."
+	var/on_store_name = "Elevator Announcer"
+
+/obj/machinery/button/remote/elevator_panel/New()
+	announce = new /obj/item/device/radio/intercom(src)
+	update_icon()
+	..()
+
+/obj/machinery/button/remote/elevator_panel/trigger()
+	var/elevator_prompt
+	var/turf/T = get_turf(src)
+
+	if(world.time - cooldown_timer > cooldown)
+		cooldown_timer = world.time
+		elevator_prompt = alert(src, "Warning! Using the elevator will end your shift and remove you from the round. Are you -sure- you want to head to your dorm for the day?\n(Everyone inside the elevator with you will end their shift and be removed as well!)", "End your shift and head to the dorms?", "End shift", "Stay on shift")
+		if(elevator_prompt == "End shift")
+			for(var/obj/machinery/door/airlock/D in SSmachines.machinery)
+				if (D.id_tag == id)
+					D.close()
+					sleep(11)
+					D.lock()
+			sleep(10)	//Wait 1 seconds before proceeding
+			playsound(src, 'zzzz_modular_occulus/sound/effects/elevator_going_up.ogg', 70, 0)
+			sleep(10)
+			visible_message("<span class='notice'>The elevator hums and creaks as it makes its way to the dormitory floor.</span>")
+			sleep(185)	//Wait 6 seconds before proceeding
+			for (var/mob/living/mobsgoingup in get_area(src))
+				despawn_passenger(mobsgoingup)
+		else
+			T.audible_message("<font color = #F63812><B>Elevator Coms</B>: \"Error. The elevator is currently out of service. Please contact your local engineering department.\"</font>")
+	else
+		to_chat(usr, SPAN_DANGER("The elevator is currently in use. Please wait before you can use it again."))
+
+/obj/machinery/button/remote/elevator_call_button
+	name = "Elevator Call Button"
+	desc = "An button used to call the elevator down to the current floor."
+	var/cooldown = 24 SECONDS
+	var/cooldown_timer
+
+/obj/machinery/button/remote/elevator_call_button/trigger()
+	for(var/obj/machinery/door/airlock/D in SSmachines.machinery)
+		if (D.id_tag == id)
+			if(D.locked)
+				if(world.time - cooldown_timer > cooldown)
+					cooldown_timer = world.time
+					visible_message("<span class='notice'>The elevator has been called. Please wait...</span>")
+					playsound(src, 'zzzz_modular_occulus/sound/effects/elevator_going_down.ogg', 70, 0)
+					sleep(220)
+					for(D in SSmachines.machinery)
+						if (D.id_tag == id)
+							D.unlock()
+							sleep(10)
+							D.open()
```

Comment:
You can use `D.do_command("secure_open")` to open and bolt the airlock open, though if you just want to have it unbolted (and not bolted open), this should be fine.

Reply:
Yeah the idea is for the airlock to act like real elevators. So once it reaches the floor it will unbolt and open, then after some time it closes. Clicking the button outside the elevator while it is unbolted will open the airlock, like normal elevator buttons do.